### PR TITLE
Pin versions for release0.59

### PR DESCRIPTION
**Pull request by sklam**
as titled

### CODE REVIEW

File: docs/source/user/installing.rst [hunk 1]
```diff
@@ -262,7 +262,7 @@ information.
 +----------++--------------+---------------------------+----------------------------+------------------------------+-------------------+-----------------------------+
 | Numba     | Release date | Python                    | NumPy                      | llvmlite                     | LLVM              | TBB                         |
 +===========+==============+===========================+============================+==============================+===================+=============================+
-| 0.59.0    | UNRELEASED   | 3.9.x <= version < 3.13   | 1.22 <= version < 1.27     | 0.42.x                       | 14.x              | 2021.6 <= version           |
+| 0.59.0    | 2023-12-14   | 3.9.x <= version < 3.13   | 1.22 <= version < 1.27     | 0.42.x                       | 14.x              | 2021.6 <= version           |
```
Comment (sklam) [inline docs/source/user/installing.rst:265]:
```suggestion
| 0.59.0    | UNRELEASED   | 3.9.x <= version < 3.13   | 1.22 <= version < 1.27     | 0.42.x                       | 14.x              | 2021.6 <= version           |
```

File: setup.py [hunk 2]
```diff
@@ -363,10 +364,11 @@ def check_file_at_path(path2file):
 
 packages = find_packages(include=["numba", "numba.*"])
 
-build_requires = ['numpy >={}'.format(min_numpy_build_version)]
+build_requires = ['numpy >={}<{}'.format(min_numpy_build_version,
+                                         max_numpy_run_version)]
```
Comment (stuartarchibald) [inline setup.py:368]:
```suggestion
build_requires = ['numpy >={},<{}'.format(min_numpy_build_version,
                                         max_numpy_run_version)]
```
Should there be a comma in here, there was in #8205.

File: setup.py [hunk 3]
```diff
@@ -363,10 +364,11 @@ def check_file_at_path(path2file):
...
 install_requires = [
     'llvmlite >={},<{}'.format(min_llvmlite_version, max_llvmlite_version),
-    'numpy >={}'.format(min_numpy_run_version),
+    'numpy >={}<{}'.format(min_numpy_run_version, max_numpy_run_version),
```
Comment (stuartarchibald) [inline setup.py:371]:
```suggestion
    'numpy >={},<{}'.format(min_numpy_run_version, max_numpy_run_version),
```
same question RE comma

**Review by stuartarchibald — COMMENTED** [review]
Thanks for the patch, couple of queries else looks good.

File: setup.py [hunk 4]
```diff
@@ -363,10 +364,11 @@ def check_file_at_path(path2file):
 
 packages = find_packages(include=["numba", "numba.*"])
 
-build_requires = ['numpy >={}'.format(min_numpy_build_version)]
+build_requires = ['numpy >={},<{}'.format(min_numpy_build_version,
+                                         max_numpy_run_version)]
```
Comment (stuartarchibald) [inline setup.py:368]:
```suggestion
                                          max_numpy_run_version)]
```
fix alignment?

**Review by stuartarchibald — APPROVED** [review]
Thanks for the patch. This looks good on inspection, code change is approved conditional on CI passing!

**Pull request merged**